test/Test/Password.hs

```diff
@@ -0,0 +1,26 @@
+module Test.Password where
+
+import Control.Monad.Except (MonadError)
+import Lib.App.Error (AppError (..))
+
+import Hedgehog (MonadGen, Property, forAll, property, withTests, (===))
+import Test.Tasty
+import Test.Tasty.Hedgehog
+
+import Lib.Util.Password
+
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+
+test_pwdHashVerified :: [TestTree]
+test_pwdHashVerified = pure $ testProperty "Password verification" pwdProperty
+
+pwdProperty :: Property
+pwdProperty  = property $ do
+  randomPwd <- forAll genPwd
+  whenRightM (runExceptT $ mkPasswordHash randomPwd) $ \pwdHash ->
+    verifyPassword randomPwd pwdHash === True
```

Comment:
I think it's better to use [`assert`](https://hackage.haskell.org/package/hedgehog-0.5.3/docs/Hedgehog.html#v:assert) here. But this is really minor.